### DISCUSSION TITLE
umijs/max项目中，有大佬遇到在父级切换子项目，<Router basename="/oms" />is not able to match the URL "/pms/" because it does not start with the basename, so the <Router> won't render anything.这个问题吗

### DISCUSSION THREAD

**Discussion by iswinter1993** [umijs/qiankun#3029] [Q&A]
umijs/max项目中，有大佬遇到在父级切换子项目，<Router basename="/oms" />is not able to match the URL "/pms/" because it does not start with the basename, so the <Router> won't render anything.这个问题吗
![微信图片_20241203171900](https://github.com/user-attachments/assets/da4f3025-f953-4b1d-b873-18d99c199bea)

**Comment by jinxiangqiang** [comment]
base: process.env.NODE_ENV === 'production' ? `/${name}/` : undefined,
  publicPath: process.env.NODE_ENV === 'production' ? `/${name}/` : undefined,
  在发生产时再加base和publicPath能避免这个问题，或者在qiankunProps中加入base： '/xxx/'
